cmd/kube-proxy/app/server_test.go

```diff
@@ -307,14 +307,28 @@ udpTimeoutMilliseconds: 123ms
 		metricsBindAddress string
 	}{
 		{
-			name:               "IPv4 config",
+			name:               "IPv4 config, bind address 0.0.0.0",
+			bindAddress:        "0.0.0.0",
+			clusterCIDR:        "1.2.3.0/24",
+			healthzBindAddress: "1.2.3.4:12345",
+			metricsBindAddress: "2.3.4.5:23456",
+		},
+		{
+			name:               "IPv4 config, non-zeros bind address",
 			bindAddress:        "9.8.7.6",
 			clusterCIDR:        "1.2.3.0/24",
 			healthzBindAddress: "1.2.3.4:12345",
 			metricsBindAddress: "2.3.4.5:23456",
 		},
 		{
-			name:               "IPv6 config",
+			name:               "IPv6 config, bind address with quotes: \"::\"",
+			bindAddress:        "\"::\"", // Need double quotes if bind address begins with ':'
```

Comment:
This is weird, so if a user passed it as a flag they would have to say `--flag=\"::\"` ?  Why?

Is `[::]` a valid alternative?

Reply:
The '\' s are only here in the test code to make Go insert literal quotes in the yaml string that gets created below in the line "yaml := fmt.Sprintf(...). This test case is testing the setting of bindAddress in a kube-proxy yaml config file that gets pointed to via --config=kube-proxy.cfg on the command line (vs. setting --bind-address directly on the command line, which I believe is deprecated for kube-proxy?). So in this yaml config file:
```
bindAddress: ::      # Fails
bindAddress: "::"    # Works
bindAddress: ::0    # Works
bindAddress: [::]    # Fails
bindAddress: "[::]"   #Works
I'll add a comment here and/or change the name of the test case to make it clear exactly how it needs to be set in the yaml file (and add some more test cases for the example settings listed here.